lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -218,6 +218,30 @@ public function setTemplateOptions($templateOptions)
     }
 
     /**
+     * Creates a Zend_Mime_Part attachment and attaches it to the current message scope.
+     * Attachment is automatically added to the mail object after creation.
+     *
+     * See also: http://framework.zend.com/manual/1.12/de/zend.mail.attachments.html
+     *
+     * @param  string $body
+     * @param  string $mimeType
+     * @param  string $disposition
+     * @param  string $encoding
+     * @param  string $filename A filename for the attachment
+     * @return $this
+     */
+    public function addAttachment(
+        $body,
+        $mimeType = \Zend_Mime::TYPE_OCTETSTREAM,
+        $disposition = \Zend_Mime::DISPOSITION_ATTACHMENT,
+        $encoding = \Zend_Mime::ENCODING_BASE64,
+        $filename = null
+    ) {
+        $this->message->createAttachment($body, $mimeType, $disposition, $encoding, $filename);
```

Comment:
As declared in constructor `$message` is `MessageInterface` which do not have method `createAttachment`. We can not add it now as that will be backward incompatible change